refactor noop transform to use dpk_ structures

daw3rd opened the pull request:
## Why are these changes needed?
To adapt to the new transform project structure.

## Related issue number (if any).
https://github.com/IBM/data-prep-kit/issues/774

daw3rd committed:
refactor noop transform to use dpk_ structures

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
readme and kfp Makefile for noop restructure

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
add spark support to restructured noop

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
fix import in spark noop test

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
fix comments and help text in noop

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
update doc references to noop to match restructuring

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
Update transform readme to match new transform structure

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
more project structure documentation change

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
add noop .dockerignore

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
correct make conventions target for new project structure

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
fix transform check-exists target to not print errors when dir does not exist

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
fix check-exists target and update .dockerignores

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
remove duplicate convention check on spark file

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
readme typo on running transform

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
rename all noop runtime files to runtime.py

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
update transform dir README with dir tree

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
update Makefile to use runtime.py for noop

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
formatting in transform readme

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
remove notebook from transform readme

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
typo in transforms readme

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
Merge branch 'dev' into noop-refactor

daw3rd committed:
fix typo in quickstart

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
set image runtime module for noop Makefile to runtime

Signed-off-by: David Wood <dawood@us.ibm.com>

shahrokhDaijavad commented:
@daw3rd I have reviewed 2 things so far:
1) Going over your README file
2) Testing the code by running `make run-cli-sample` successfully

On the README, everything looks good, but I have a suggestion for improvement:
In the Usage section, both CLI and `make run-cli-sample` use the Python runtime. Since noop is a good learning transform for first-time users, it makes sense to add two small sections of code for using CLI to run Ray and Spark runtimes. An example is the README for fdedup here: https://github.com/IBM/data-prep-kit/blob/noop-refactor/transforms/universal/fdedup/README.md. 
Since we have the generic: python -m dpk_$(TRANSFORM_NAME).runtime
an alternative, if you do not want to have new Ray and Spark sections in the README,  is just changing the CLI part to:
python -m dpk_noop/transform_python --noop_sleep_sec 10 ... 
=>
python -m dpk_noop/transform_runtime --noop_sleep_sec 10 ... and saying that runtime can be python, ray, or spark.

daw3rd committed:
transform readme changes

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
change noop module to runtime in kfp_ray

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
adapt to name change of module to runtime in tests and readme for noop

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
fix noop module references in docs to match new runtime module naming

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd commented:
> @daw3rd I have reviewed 2 things so far:
> 
>     1. Going over your README file
> 
>     2. Testing the code by running `make run-cli-sample` successfully
> 
> 
> On the README, everything looks good, but I have a suggestion for improvement: In the Usage section, both CLI and `make run-cli-sample` use the Python runtime. Since noop is a good learning transform for first-time users, it makes sense to add two small sections of code for using CLI to run Ray and Spark runtimes. An example is the README for fdedup here: https://github.com/IBM/data-prep-kit/blob/noop-refactor/transforms/universal/fdedup/README.md. Since we have the generic: python -m dpk_$(TRANSFORM_NAME).runtime an alternative, if you do not want to have new Ray and Spark sections in the README, is just changing the CLI part to: python -m dpk_noop/transform_python --noop_sleep_sec 10 ... => python -m dpk_noop/transform_runtime --noop_sleep_sec 10 ... and saying that runtime can be python, ray, or spark.

I added an example of running the Ray runtime. Do we need to show how to run all the local_*.py?

daw3rd committed:
fix typo in noop readme

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
more noop renamings in markdown

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
add README.md.template for transforms

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
transform readme updates

Signed-off-by: David Wood <dawood@us.ibm.com>

shahrokhDaijavad commented:
> > @daw3rd I have reviewed 2 things so far:
> > ```
> > 1. Going over your README file
> > 
> > 2. Testing the code by running `make run-cli-sample` successfully
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > On the README, everything looks good, but I have a suggestion for improvement: In the Usage section, both CLI and `make run-cli-sample` use the Python runtime. Since noop is a good learning transform for first-time users, it makes sense to add two small sections of code for using CLI to run Ray and Spark runtimes. An example is the README for fdedup here: https://github.com/IBM/data-prep-kit/blob/noop-refactor/transforms/universal/fdedup/README.md. Since we have the generic: python -m dpk_$(TRANSFORM_NAME).runtime an alternative, if you do not want to have new Ray and Spark sections in the README, is just changing the CLI part to: python -m dpk_noop/transform_python --noop_sleep_sec 10 ... => python -m dpk_noop/transform_runtime --noop_sleep_sec 10 ... and saying that runtime can be python, ray, or spark.
> 
> I added an example of running the Ray runtime. Do we need to show how to run all the local_*.py?

@daw3rd We don't need to show how to run all 3, but when you show the command line for Ray runtime, you want to add a sentence saying that the runtime could be python, ray or spark.

touma-I commented on data-processing-lib/doc/python-launcher-options.md:
````diff
@@ -6,7 +6,7 @@ The following is a current --help output (a work in progress) for
 
 ```
-usage: noop_python_runtime.py [-h] [--noop_sleep_sec NOOP_SLEEP_SEC] [--noop_pwd NOOP_PWD] [--data_s3_cred DATA_S3_CRED] [--data_s3_config DATA_S3_CONFIG] [--data_local_config DATA_LOCAL_CONFIG] [--data_max_files DATA_MAX_FILES]
+usage: dpk_noop/runtime.py [-h] [--noop_sleep_sec NOOP_SLEEP_SEC] [--noop_pwd NOOP_PWD] [--data_s3_cred DATA_S3_CRED] [--data_s3_config DATA_S3_CONFIG] [--data_local_config DATA_LOCAL_CONFIG] [--data_max_files DATA_MAX_FILES]
````
This is confusing. It should be: python -m dpk_noop.runtime [-h] etc...

daw3rd replied:
Here is what --help spits outs (runtime.py). So maybe I need to remove `dpk_noop`.
```
(venv) dawood@davids-mbp:~/git2/data-prep-kit/transforms/universal/noop$ python -m dpk_noop.runtime --help
10:41:39 INFO - Launching noop transform
usage: runtime.py [-h] [--noop_sleep_sec NOOP_SLEEP_SEC] [--noop_pwd NOOP_PWD] [--data_s3_cred DATA_S3_CRED]
                  [--data_s3_config DATA_S3_CONFIG] [--data_local_config DATA_LOCAL_CONFIG] [--data_max_files DATA_MAX_FILES]
                  [--data_checkpointing DATA_CHECKPOINTING] [--data_files_to_checkpoint DATA_FILES_TO_CHECKPOINT]
                  [--data_data_sets DATA_DATA_SETS] [--data_files_to_use DATA_FILES_TO_USE] [--data_num_samples DATA_NUM_SAMPLES]
                  [--runtime_num_processors RUNTIME_NUM_PROCESSORS] [--runtime_pipeline_id RUNTIME_PIPELINE_ID]
                  [--runtime_job_id RUNTIME_JOB_ID] [--runtime_code_location RUNTIME_CODE_LOCATION]

```

daw3rd replied:
I updated to match the above.

daw3rd replied:
Will update to create a single doc with options and no noop.

daw3rd replied:
Done

touma-I commented on data-processing-lib/doc/python-launcher-options.md:
Does this belong here or should it be in noop folder and linked to it from here ? is there another way where we can explain the runtime configuration parameters without getting it entangled with the operator ?

daw3rd replied:
That is a different PR :)

touma-I commented on data-processing-lib/doc/ray-launcher-options.md:
````diff
@@ -5,7 +5,7 @@ The following is a current --help output (a work in progress) for
 
 ```
-usage: noop_transform.py [-h] [--run_locally RUN_LOCALLY] [--noop_sleep_sec NOOP_SLEEP_SEC] [--noop_pwd NOOP_PWD] [--data_s3_cred DATA_S3_CRED] [--data_s3_config DATA_S3_CONFIG] [--data_local_config DATA_LOCAL_CONFIG]
+usage: dpk_noop/ray/runtime.py [-h] [--run_locally RUN_LOCALLY] [--noop_sleep_sec NOOP_SLEEP_SEC] [--noop_pwd NOOP_PWD] [--data_s3_cred DATA_S3_CRED] [--data_s3_config DATA_S3_CONFIG] [--data_local_config DATA_LOCAL_CONFIG]
````
should be: python -m dpk_noop.ray.runtime [-h] etc..

daw3rd replied:
Same comment as above

daw3rd replied:
Will update to create a single doc with options and no noop.

daw3rd replied:
Done

touma-I commented on data-processing-lib/doc/simplified_transform_apis.md:
Do we still need this ?

daw3rd replied:
@shahrokhDaijavad to remove

daw3rd replied:
Done

touma-I commented on data-processing-lib/doc/spark-launcher-options.md:
````diff
@@ -7,7 +7,7 @@ The following is a current --help output (a work in progress) for
 
 ```
-usage: noop_python_runtime.py [-h] [--noop_sleep_sec NOOP_SLEEP_SEC] [--noop_pwd NOOP_PWD] [--data_s3_cred DATA_S3_CRED] [--data_s3_config DATA_S3_CONFIG] [--data_local_config DATA_LOCAL_CONFIG] [--data_max_files DATA_MAX_FILES]
+usage: noop_dpk/spark/runtime.py [-h] [--noop_sleep_sec NOOP_SLEEP_SEC] [--noop_pwd NOOP_PWD] [--data_s3_cred DATA_S3_CRED] [--data_s3_config DATA_S3_CONFIG] [--data_local_config DATA_LOCAL_CONFIG] [--data_max_files DATA_MAX_FILES]
````
Something does not look right. I expected:
python -m dpk_noop.spark.runtime [-h] ...

daw3rd replied:
Same comment as above.

daw3rd replied:
Will update to create a single doc with options and no noop.

daw3rd replied:
Done

touma-I commented on data-processing-lib/doc/transform-standalone-testing.md:
```diff
@@ -62,10 +62,10 @@ to use different models and perhaps as a result have different results.
 % make venv
 % source venv/bin/activate
-(venv)% export PYTHONPATH=.../data-prep-kit/transforms/universal/noop/src
+(venv)% export PYTHONPATH=.
```
why is this needed? The current folder should be "implicitly" added to the python path. need to test without it.

daw3rd replied:
Yes, this fails if we don't set PYTHONPATH.  Something specific to pytest perhaps.

touma-I commented on data-processing-lib/doc/transform-standalone-testing.md:
Note to self: need to test all the scripts discussed here...

touma-I commented on data-processing-lib/doc/transform-standalone-testing.md:
```diff
@@ -82,7 +82,7 @@ Finally, the tests should be runnable with the `Makefile`  as follows:
 $ make test
 source venv/bin/activate;			\
-	export PYTHONPATH=../src:.:$PYTHONPATH;	\
+	export PYTHONPATH=..:.:$PYTHONPATH;	\
```
does not seem right. Need to double check this...

daw3rd replied:
It is needed, but I updated the file to not show the details of `make test`

daw3rd replied:
Also removed export PYTHONPATH from previous example.

touma-I commented on doc/quick-start/contribute-your-own-transform.md:
Nice Catch! :-)

touma-I commented on doc/quick-start/run-transform-image.md:
```diff
@@ -66,15 +67,15 @@ docker run  --rm
 To use the ray runtime, we must 
 1. Switch to using the ray-based image `noop-ray:latest`
-2. Use the ray runtime python main() defined in `noop_transform_ray.py`
+2. Use the ray runtime python main() defined in `dpk_noop/ray/runtime.py`
```
use `__main__` instead of `main()`

daw3rd replied:
Done

touma-I commented on doc/repo.md:
I don't think Developer should be able to publish ? Both pypi and quay have protected access.

daw3rd replied:
Removed

daw3rd committed:
noop readme and transform readme template

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd commented:
> > > @daw3rd I have reviewed 2 things so far:
> > > ```
> > > 1. Going over your README file
> > > 
> > > 2. Testing the code by running `make run-cli-sample` successfully
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > On the README, everything looks good, but I have a suggestion for improvement: In the Usage section, both CLI and `make run-cli-sample` use the Python runtime. Since noop is a good learning transform for first-time users, it makes sense to add two small sections of code for using CLI to run Ray and Spark runtimes. An example is the README for fdedup here: https://github.com/IBM/data-prep-kit/blob/noop-refactor/transforms/universal/fdedup/README.md. Since we have the generic: python -m dpk_$(TRANSFORM_NAME).runtime an alternative, if you do not want to have new Ray and Spark sections in the README, is just changing the CLI part to: python -m dpk_noop/transform_python --noop_sleep_sec 10 ... => python -m dpk_noop/transform_runtime --noop_sleep_sec 10 ... and saying that runtime can be python, ray, or spark.
> > 
> > 
> > I added an example of running the Ray runtime. Do we need to show how to run all the local_*.py?
> 
> @daw3rd We don't need to show how to run all 3, but when you show the command line for Ray runtime, you want to add a sentence saying that the runtime could be python, ray or spark.

@shahrokhDaijavad OK I added spark example and was more specific about running the various runtimes.  If you want more, maybe get together to discuss.

shahrokhDaijavad commented:
Thanks, @daw3rd Looks good!

touma-I reviewed (commented):
trying to go through this with a fine comb as the noop is a reference for everything we do. will have some more tomorrow during our call. thanks

touma-I commented on data-processing-lib/doc/simplified_transform_apis.md:
Should we better explain how one would invoke the local_python, local_ray and local_spark using the -m option ?

daw3rd replied:
@shahrokhDaijavad to remove

daw3rd replied:
done

touma-I commented on doc/repo.md:
II don't think the developer can publish!

daw3rd replied:
Removed (duplicate of above?)

touma-I commented on transforms/universal/noop/Makefile:
```diff
@@ -1,79 +1,30 @@
-	@# Help: Recursively make $@ all subdirs 
-	$(MAKE) RULE=$@ .recurse
+# Until we make runtime.py the standard supported by Makefile infra
+TRANSFORM_PYTHON_SRC="-m dpk_$(TRANSFORM_NAME).runtime"
```
Can you add those to the template ?

daw3rd replied:
Done

touma-I commented on transforms/universal/noop/dpk_noop/local_python.py:
```diff
@@ -15,7 +15,7 @@
 from data_processing.runtime.pure_python import PythonTransformLauncher
 from data_processing.utils import ParamsUtils
-from noop_transform_python import NOOPPythonTransformConfiguration
+from dpk_noop.transform_python import NOOPPythonTransformConfiguration
```
Did we change transform_python to runtime ?

daw3rd replied:
Nice catch. Fixed and now runs with success.

touma-I commented on transforms/universal/noop/dpk_noop/local_python_multiprocessor.py:
```diff
@@ -15,7 +15,7 @@
 from data_processing.runtime.pure_python import PythonTransformLauncher
 from data_processing.utils import ParamsUtils
-from noop_transform_python import NOOPPythonTransformConfiguration
+from dpk_noop.transform_python import NOOPPythonTransformConfiguration
```
```suggestion
from dpk_noop.runtime import NOOPPythonTransformConfiguration
```

daw3rd replied:
Nice catch. Fixed and now runs with success.

touma-I commented on transforms/universal/noop/dpk_noop/ray/local_ray.py:
```diff
@@ -15,7 +15,7 @@
 from data_processing.utils import ParamsUtils
 from data_processing_ray.runtime.ray import RayTransformLauncher
-from noop_transform_ray import NOOPRayTransformConfiguration
+from dpk_noop.ray.transform import NOOPRayTransformConfiguration
```
```suggestion
from dpk_noop.ray.runtime import NOOPRayTransformConfiguration
```

daw3rd replied:
Nice catch. Fixed and now runs with success.

touma-I commented on transforms/universal/noop/dpk_noop/ray/s3_ray.py:
```diff
@@ -15,7 +15,7 @@
 from data_processing.utils import ParamsUtils
 from data_processing_ray.runtime.ray import RayTransformLauncher
-from noop_transform_ray import NOOPRayTransformConfiguration
+from dpk_noop.ray.transform import NOOPRayTransformConfiguration
```
```suggestion
from dpk_noop.ray.runtime import NOOPRayTransformConfiguration
```

daw3rd replied:
Nice catch. Fixed

touma-I commented on transforms/universal/noop/dpk_noop/spark/local_spark.py:
```diff
@@ -15,7 +15,7 @@
 from data_processing.utils import ParamsUtils
 from data_processing_spark.runtime.spark import SparkTransformLauncher
-from noop_transform_spark import NOOPSparkTransformConfiguration
+from dpk_noop.spark.transform import NOOPSparkTransformConfiguration
```
```suggestion
from dpk_noop.spark.runtime import NOOPSparkTransformConfiguration
```

daw3rd replied:
Nice catch. Fixed and now runs with success.

touma-I reviewed (commented):
more comments in preparation for our call. thanks

daw3rd committed:
address doc comments, fix local_*.py to reference correct package and test-data dir

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
fix noop s3 import

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
edits to launcher md to only show runtime.py

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
remove details on  execution from md

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
make runtime.py the default SRC in the transform Makefile template

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
replace python/ray/spark launcher options docs with a single doc on launcher options

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
fix testing doc by removing setting of PYTHONPATH

Signed-off-by: David Wood <dawood@us.ibm.com>

shahrokhDaijavad committed:
Modifications/Deletions of a few md files

Signed-off-by: SHAHROKH DAIJAVAD <shahrokh@us.ibm.com>

touma-I commented on data-processing-lib/doc/overview.md:
```diff
@@ -45,8 +45,7 @@ To learn more consider the following:
 * [Transform Runtimes](transform-runtimes.md)
-* [Transform Examples](transform-tutorial-examples.md)
-* [Simplified transform APIs](simplified_transform_apis.md)
+* [Transform Tutorial Example](transform-tutorial-examples.md)
```
Change to point to: quick-start/contribute-your-own-transform.md
```suggestion
* [Transform Tutorial Example](quick-start/contribute-your-own-transform.md)
```

touma-I committed:
Update data-processing-lib/doc/overview.md

touma-I commented on data-processing-lib/doc/overview.md:
```diff
@@ -45,8 +45,7 @@ To learn more consider the following:
 * [Transform Runtimes](transform-runtimes.md)
-* [Transform Examples](transform-tutorial-examples.md)
-* [Simplified transform APIs](simplified_transform_apis.md)
+* [Transform Tutorial Example](quick-start/contribute-your-own-transform.md)
```
```suggestion
* [Transform Tutorial Example](../../doc/quick-start/contribute-your-own-transform.md)
```

touma-I committed:
Update data-processing-lib/doc/overview.md
